server/routes/applications.js

```diff
@@ -295,6 +295,12 @@ router.post('/manage-applications/:appId/:keyType/:keyId/update/:env', (req, res
   const data = req.body;
   const viewData = routeUtils.createViewData('Update Key', 'application-overview', req);
   viewData.this_data = data;
+  viewData.this_data = {
+    appId: appId,
+    env: env,
+    keyType: keyType,
+    keyId: keyId
+  };
```

Comment:
I'd move all this into the catch and delete the `viewData.this_data = data` since its being overwritten anyway

Reply:
I had to change it so that it wan't being overwritten because the data contained the request body and needed to be persisted as well. But I have moved the relevant assignments into the catch block